### PR TITLE
Add filter for RQA and follow ups survey messages labelled as Noise_O…

### PR DESCRIPTION
**Pull request by IsaackMwenda**
…ther_Project

### CODE REVIEW

File: src/filter_noise_other_project.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from src.lib import MessageFilters
+
+
+class FilterNOP(object):
+    @staticmethod
+    def filter_noise_other_project(messages):
+        """
+        Filters a list of  RQA and followup survey messages which have been labelled as Noise_Other_Project(NOP).
```
Comment (as2388) [inline src/filter_noise_other_project.py]: Could probably be slightly clearer here in explaining that this is filtering out messages which have that label

File: src/lib/message_filters.py [hunk 2]
```diff
@@ -21,3 +28,30 @@ def filter_empty_messages(messages, message_keys):
...
+                if plan.binary_coded_field is not None and td[plan.binary_coded_field]["CodeID"] != plan.code_scheme.get_code_with_control_code(Codes.NOISE_OTHER_PROJECT) and \
+                        td[plan.binary_coded_field]["Checked"]:
+                    not_noise.append(td)
+
+                # Checking for NOP label in datasets that don't have a binary scheme therefore it was labelled in normal scheme
+                if plan.binary_coded_field is None and td[plan.coded_field][0]["CodeID"] != plan.code_scheme.get_code_with_control_code(Codes.NOISE_OTHER_PROJECT) and \
+                        td[plan.coded_field][0]["Checked"]:
+                    not_noise.append(td)
+
+        log.info(f"Filtered out noise messages. "
```
Comment (as2388) [inline src/lib/message_filters.py]:
```suggestion
        log.info(f"Filtered out noise (other project) messages. "
```

Comment (as2388) [reply src/lib/message_filters.py]: (we have other forms of noise)

File: src/lib/message_filters.py [hunk 3]
```diff
@@ -21,3 +28,30 @@ def filter_empty_messages(messages, message_keys):
...
+        :rtype: list of TracedData
+        """
+        not_noise = []
+        for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+            for td in messages:
+                if plan.binary_coded_field is not None and td[plan.binary_coded_field]["CodeID"] != plan.code_scheme.get_code_with_control_code(Codes.NOISE_OTHER_PROJECT) and \
+                        td[plan.binary_coded_field]["Checked"]:
+                    not_noise.append(td)
+
+                # Checking for NOP label in datasets that don't have a binary scheme therefore it was labelled in normal scheme
```
Comment (as2388) [inline src/lib/message_filters.py]: (this is a bit yucky but realise there is nothing you can do until we redesign the relevant parts of configuration)

File: src/lib/message_filters.py [hunk 4]
```diff
@@ -21,3 +28,30 @@ def filter_empty_messages(messages, message_keys):
...
+        :param messages: List of message objects to filter.
+        :type messages: list of TracedData
+        :return: Filtered list.
+        :rtype: list of TracedData
+        """
+        not_noise = []
+        for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+            for td in messages:
+                if plan.binary_coded_field is not None and td[plan.binary_coded_field]["CodeID"] != plan.code_scheme.get_code_with_control_code(Codes.NOISE_OTHER_PROJECT) and \
+                        td[plan.binary_coded_field]["Checked"]:
```
Comment (as2388) [inline src/lib/message_filters.py]: Do you need to check this field? Thought that the pipeline would set "NR" if not checked?

**Review by as2388 — APPROVED** [review]

File: src/filter_noise_other_project.py (hunk 1 above)
Comment (IsaackMwenda) [reply src/filter_noise_other_project.py]: Updated thanks

File: src/lib/message_filters.py (hunk 2 above)
Comment (IsaackMwenda) [reply src/lib/message_filters.py]: updated thanks

**Pull request merged**